src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/converter/KeyConverter.java

```diff
@@ -43,19 +45,46 @@
 
     private final TokenCertificateConverter tokenCertificateConverter;
     private final TokenCertificateSigningRequestConverter tokenCsrConverter;
+    private final StateChangeActionHelper stateChangeActionHelper;
+    private final StateChangeActionConverter stateChangeActionConverter;
 
     @Autowired
     public KeyConverter(TokenCertificateConverter tokenCertificateConverter,
-            TokenCertificateSigningRequestConverter tokenCsrConverter) {
+            TokenCertificateSigningRequestConverter tokenCsrConverter,
+            StateChangeActionHelper stateChangeActionHelper,
+            StateChangeActionConverter stateChangeActionConverter) {
         this.tokenCertificateConverter = tokenCertificateConverter;
         this.tokenCsrConverter = tokenCsrConverter;
+        this.stateChangeActionHelper = stateChangeActionHelper;
+        this.stateChangeActionConverter = stateChangeActionConverter;
     }
 
     /**
      * Convert {@link KeyInfo} to openapi {@link Key} object
      * @param keyInfo
      */
     public Key convert(KeyInfo keyInfo) {
+        return convertInternal(keyInfo, null);
+    }
+
+    /**
+     * Convert {@link KeyInfo} to openapi {@link Key} object
+     * and populate possibleActions
+     * @param keyInfo
+     */
+    public Key convert(KeyInfo keyInfo, TokenInfo tokenInfo) {
+        if (tokenInfo == null) {
+            throw new NullPointerException("tokenInfo is mandatory to populate possibleActions");
```

Comment:
Should this be `IllegalArgumentException` instead? I think seeing a `NullPointerException` could easily be interpreted as a bug in the system.

Reply:
Good question. I researched a bit and `IllegalArgumentException` vs `NullPointerException` for null argument seems to be opinion-based, e.g. https://stackoverflow.com/questions/3881/illegalargumentexception-or-nullpointerexception-for-a-null-parameter/8196334#8196334

Accepted answer speaks for IllegalArgumentException but there are some good arguments for use of NPE, such as use in JDK libraries and Guava. 

I think both are valid choices and neither is clearly wrong, but we should be consistent. I did not find other cases from proxy-ui-api modules, but other modules seem to use IllegalArgumentException, so I will change exception type to that. And let's use that as a convention from now on.

Reply:
fixed